core/store/src/trie/mod.rs

```diff
@@ -948,7 +948,7 @@ mod tests {
         let root = test_populate_trie(&tries, &Trie::EMPTY_ROOT, ShardUId::single_shard(), changes);
         let trie = tries.get_trie_for_shard(ShardUId::single_shard(), root.clone());
         let mut iter = trie.iter().unwrap();
-        iter.seek(&vec![0, 116, 101, 115, 116, 44]).unwrap();
+        iter.seek_prefix(&vec![0, 116, 101, 115, 116, 44]).unwrap();
```

Comment:
It fails

Reply:
will need to create a new element

Reply:
added on 591a2e4958a1e64993247ee2a66e4aa36d9403d0

Reply:
No, that’s wrong.  It should pass without that element.  It doesn’t because the seek_prefix implementation isn’t correct.  Similarly, `assert!(key.starts_with(&query),` mentioned in another comment fails the test because the current implementation is incorrect.